store/tikv/gc_worker.go

```diff
@@ -43,10 +45,110 @@ type GCWorker struct {
 	lastFinish  time.Time
 	cancel      goctx.CancelFunc
 	done        chan error
+
+	session tidb.Session
+}
+
+// SafePointKV is used for a seamingless integration for mockTest and runtime.
+type SafePointKV interface {
+	Put(k string, v string) error
+	Get(k string) (string, error)
+}
+
+// MockSafePointKV implements SafePointKV at mock test
+type MockSafePointKV struct {
+	store    map[string]string
+	mockLock sync.RWMutex
+}
+
+// NewMockSafePointKV creates an instance of MockSafePointKV
+func NewMockSafePointKV() *MockSafePointKV {
+	return &MockSafePointKV{
+		store: make(map[string]string),
+	}
+}
+
+// Put implements the Put method for SafePointKV
+func (w *MockSafePointKV) Put(k string, v string) error {
+	w.mockLock.Lock()
+	defer w.mockLock.Unlock()
+	w.store[k] = v
+	return nil
+}
+
+// Get implements the Get method for SafePointKV
+func (w *MockSafePointKV) Get(k string) (string, error) {
+	w.mockLock.RLock()
+	defer w.mockLock.RUnlock()
+	elem, ok := w.store[k]
+	if !ok {
+		return "", nil
+	}
+	return elem, nil
+}
+
+// EtcdSafePointKV implements SafePointKV at runtime
+type EtcdSafePointKV struct {
+	cli *clientv3.Client
+}
+
+// NewEtcdSafePointKV creates an instance of EtcdSafePointKV
+func NewEtcdSafePointKV(addrs []string) (*EtcdSafePointKV, error) {
+	etcdCli, err := createEtcdKV(addrs)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return &EtcdSafePointKV{cli: etcdCli}, nil
+}
+
+// Put implements the Put method for SafePointKV
+func (w *EtcdSafePointKV) Put(k string, v string) error {
+	ctx, cancel := goctx.WithTimeout(goctx.Background(), time.Second*5)
+	_, err := w.cli.Put(ctx, k, v)
+	cancel()
+	if err != nil {
+		return errors.Trace(err)
+	}
+	return nil
+}
+
+// Get implements the Get method for SafePointKV
+func (w *EtcdSafePointKV) Get(k string) (string, error) {
+	ctx, cancel := goctx.WithTimeout(goctx.Background(), time.Second*5)
+	resp, err := w.cli.Get(ctx, k)
+	cancel()
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+	if len(resp.Kvs) > 0 {
+		return string(resp.Kvs[0].Value), nil
+	}
+	return "", nil
+}
+
+// StartSafePointChecker triggers SafePoint Checker on each tidb
+func (w *GCWorker) StartSafePointChecker() {
+	go func() {
+		ticker := time.NewTicker(gcSafePointUpdateInterval)
+		defer ticker.Stop()
+		for {
+			select {
+			case spCachedTime := <-ticker.C:
+				cachedSafePoint, err := w.loadSafePoint(gcSavedSafePoint)
+				if err == nil {
+					w.store.UpdateSPCache(cachedSafePoint, spCachedTime)
+				} else {
+					log.Error("The read fails:", errors.ErrorStack(err))
```

Comment:
I think we should at least retry 3~5 times if error happens, otherwise `spCachedTime` won't be updated for at least 10s,
during which `CheckVisibility` would fail and service is unavailable.

If `CheckVisibility` fail, would us backoff that operation? @disksing 
And if `spCachedTime` is not updated, backoff won't work.